test/gtest/uct/uct_test.cc

```diff
@@ -334,17 +334,42 @@ uct_test::entity::entity(const resource& resource, uct_iface_config_t *iface_con
 
 
 void uct_test::entity::mem_alloc(size_t length, uct_allocated_memory_t *mem,
-                                 uct_rkey_bundle *rkey_bundle) const {
+                                 uct_rkey_bundle *rkey_bundle, int mem_type) const {
     static const char *alloc_name = "uct_test";
     ucs_status_t status;
     void *rkey_buffer;
 
     if (md_attr().cap.flags & (UCT_MD_FLAG_ALLOC|UCT_MD_FLAG_REG)) {
-        status = uct_iface_mem_alloc(m_iface, length, UCT_MD_MEM_ACCESS_ALL,
-                                     alloc_name, mem);
-        ASSERT_UCS_OK(status);
+        if (mem_type == UCT_MD_MEM_TYPE_HOST) {
+            status = uct_iface_mem_alloc(m_iface, length, UCT_MD_MEM_ACCESS_ALL,
+                                         alloc_name, mem);
+            ASSERT_UCS_OK(status);
+        } else if(mem_type == UCT_MD_MEM_TYPE_CUDA) {
+#if HAVE_CUDA
+            cudaError_t cerr;
```

Comment:
maybe extract the code which allocates cuda mem and registers it on a memory domain to a helper function?